sklearn/decomposition/nmf.py

```diff
@@ -277,30 +632,208 @@ class ProjectedGradientNMF(BaseEstimator, TransformerMixin):
+        raise ValueError("Number of components must be positive;"
+                         " got (n_components=%r)" % n_components)
+    if not isinstance(max_iter, numbers.Number) or max_iter < 0:
+        raise ValueError("Maximum number of iteration must be positive;"
+                         " got (max_iter=%r)" % max_iter)
+    if not isinstance(tol, numbers.Number) or tol < 0:
+        raise ValueError("Tolerance for stopping criteria must be "
+                         "positive; got (tol=%r)" % tol)
+
+    # check W and H, or initialize them
+    if init == 'custom':
+        _check_init(H, (n_components, n_features), "NMF (input H)")
+        _check_init(W, (n_samples, n_components), "NMF (input W)")
+    elif not update_H:
+        _check_init(H, (n_components, n_features), "NMF (input H)")
+        W = np.zeros((n_samples, n_components))
+    else:
+        W, H = _initialize_nmf(X, n_components, init,
+                               random_state=random_state)
+
+    if solver == 'proj-grad':
+        warnings.warn("'proj-grad' solver will be removed in release 0.19."
+                      " Use 'coordinate' solver instead.", DeprecationWarning)
+        if update_H:  # fit_transform
+            W, H, n_iter = _fit_projected_gradient(X, W, H, tol,
+                                                   max_iter,
+                                                   nls_max_iter,
+                                                   alpha, l1_ratio,
+                                                   sparseness,
+                                                   beta, eta)
+        else:  # transform
+            W, H, n_iter = _update_projected_gradient_w(X, W, H,
+                                                        tol, nls_max_iter,
+                                                        alpha, l1_ratio,
+                                                        sparseness, beta,
+                                                        eta)
+    elif solver == 'coordinate':
+        W, H, n_iter = _fit_coordinate_descent(X, W, H, tol,
+                                               max_iter,
+                                               alpha, l1_ratio,
+                                               regularization,
+                                               update_H=update_H,
+                                               verbose=verbose,
+                                               shuffle=shuffle,
+                                               random_state=random_state)
+    else:
+        raise ValueError("Invalid solver parameter '%s'." % solver)
+
+    if n_iter == max_iter:
+        warnings.warn("Maximum number of iteration %d reached. Increase it to"
+                      " improve convergence." % max_iter, ConvergenceWarning)
+
+    return W, H, n_iter
+
+
+class NMF(BaseEstimator, TransformerMixin):
+    """Non-Negative matrix factorization (NMF)
+
```

Comment:
capitalize M and F. 

I would suggest making this docstring a bit longer. A sentence explaining the model and applications, then the loss functions, then the link to the user guide.